pysat/_instrument.py

```diff
@@ -291,7 +334,21 @@ def __init__(self, platform=None, name=None, tag=None, inst_id=None,
                                fill_label=self.fill_label)
 
         # function processing class, processes data on load
-        self.custom = pysat.Custom()
+        self.custom_functions = []
+        self.custom_kind = []
+        self.custom_args = []
+        self.custom_kwargs = []
+        if custom is not None:
+            # process user input
+            req_labels = ['function', 'kind', 'args', 'kwargs']
```

Comment:
```suggestion
            req_keys = ['function', 'kind']
            opt_keys = ['args', 'kwargs']
```

After so much time in Meta using `labels` here seems less than ideal, though not against the style guidelines.  My other thought is "why not make args and kwargs optional?"  It'd be easy to add an empty list or dict if its missing and many routines don't need both or even one of them.